### ISSUE TITLE
[Bug]: Appsmith hangs, nothing works [Screen recording]

### ISSUE THREAD

**Issue by codenameakshay** [appsmithorg/appsmith#30283]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Any change I do, I get a blank error toast, and any datasource or API call just keeps on loading.

### Steps To Reproduce

Happening only on a branch, not on master. I think it is specific to my repository, don't think it is reproducible.

### Environment

Production

### Issue video log

https://drive.google.com/file/d/1m2ukhxj-lFwvkirUn5wLg8rtRqteHHqI/view?usp=sharing

### Version

Self Hosted - Appsmith v1.9.56

**Comment by Nikhil-Nandagopal** [comment]
@codenameakshay can you check whether there's a cyclic dependency between the functions you are calling? It's possible that this may be leading to an infinite loop

**Comment by codenameakshay** [comment]
@Nikhil-Nandagopal I checked, there is no cyclic dependency between the functions. In fact, I am simply calling an API which seems to work if created in a new page, but doesn't work in this page when copied over. I have attached more code, and how it is happening over the email thread.

**Comment by codenameakshay** [comment]
I had to create a new AppSmith project, then connect it with my git repo to fix the issue. But this was frustrating, as I had to setup the user groups' and roles' permissions for that app again. It would've been great if I had some button in AppSmith to hard pull from the remote instead of it working as a local, and then merging the changes whenever I pull. (Discard and pull is kind of soft, as it only removes the unstaged changes and not the commits which are not present in remote. I cannot also change default branch to something else and then delete main, and then re-pull it, as it is an enterprise level feature 🤦 )
